### PR TITLE
fix(pie-toast): DSW-2476 rtl updates to the component

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)

- Enures the alignment and animation of the component follows the design guidelines for both LTR and RTL. The component should slide from the left for LTR and from the right for RTL.
- Updates the motion illustration as requested by design. 
- Updates the padding values of the container to match designs as requested in the design review

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have added thorough tests where applicable (unit / component / visual)
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] I have reviewed visual test updates properly before approving
- [x] If changes will affect consumers of the package, I have created a changeset entry.

### Reviewer 1 - @fernandofranca 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2 - @siggerzz 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-toast/src/index.ts [hunk 1]
```diff
@@ -15,31 +15,34 @@ import {
...
 } from './defs';
 
 // Valid values available to consumers
 export * from './defs';
 
 /**
  * @tagname pie-toast
+ * @event {CustomEvent} pie-toast-close - when a user clicks close button or when the toast auto dismiss.
+ * @event {CustomEvent} pie-toast-open - when the toast is opened.
+* @event {CustomEvent} pie-toast-leading-action-click - when the user interacts with the leading action.
```
Comment (fernandofranca) [inline packages/components/pie-toast/src/index.ts:45]: nit: indentation missing a space

File: packages/components/pie-toast/src/toast.scss [hunk 2]
```diff
@@ -92,18 +106,32 @@
   display: flex;
   align-items: center;
   gap: var(--dt-spacing-b);
-  padding: calc(var(--dt-spacing-03) / 2) 0;
+  padding: 6px var(--dt-spacing-a) 6px 0;
```
Comment (fernandofranca) [inline packages/components/pie-toast/src/toast.scss:109]: Can we add a css variable to self document why this magic `6px` exists?

Comment (raoufswe) [reply packages/components/pie-toast/src/toast.scss:109]: it's per design specs and we don't have a token for it.

File: packages/components/pie-toast/src/index.ts (hunk 1 above)
Comment (raoufswe) [reply packages/components/pie-toast/src/index.ts]: done

**Review by siggerzz — APPROVED** [review]

**Review by fernandofranca — APPROVED** [review]

**Pull request merged**